src/item.cpp

```diff
@@ -5288,7 +5297,12 @@ int item::get_coverage() const
     if( t == nullptr ) {
         return 0;
     }
-    return t->coverage;
+    int result = t->coverage;
+
+    result += get_clothing_mod_val( clothing_mod_type_coverage );
+
+    // note: this may not be the best place to put this check?
+    return std::max( 0, std::min( result, 100 ) );
```

Comment:
Why the min? Why is encumbrance capped at 100?

Reply:
This is coverage. Since clothing_mod now has clothing_mod_type_coverage it's possible to create a mod that results in coverage < 0 or > 100, which is obviously wrong. Came up while testing, applying a coverage increase to a coat that already had 95% coverage.

I did not cap other values, though I wondered if having negative warmth, encumbrance, or protection should be maxed to 0 too. What happens if something has a negative final value here?

The question is whether putting it here in get_coverage() is the right way to handle it. Does anything bypass get_coverage() when doing coverage checks and access the value directly? since in the past nothing changed coverage dynamically? If so would it be better to fix those spots to always call get_coverage(), or better to handle the problem in a different way?

Reply:
Ack, my bad!

I'd say the other values should be floored at 0 as well, and this is probably the right place to stick it. If other places are accessing it directly, they should probably be converted to use this.

Reply:
Ok, I'll go ahead and do that since I'm already in there.

Reply:
Ah yeah I probably should have checked if there was a clamp function >.< Thanks. My day job is mostly javascript, which doesn't have clamp in the core library so I'm in the habit of doing this manually <.<